src/doc/src/reference/unstable.md

```diff
@@ -1665,6 +1666,96 @@ version = "0.1.0"
 workspace = true
 ```
 
+## Profile `trim-paths` option
+
+* Tracking Issue: [rust-lang/cargo#12137](https://github.com/rust-lang/cargo/issues/12137)
+* Tracking Rustc Issue: [rust-lang/rust#111540](https://github.com/rust-lang/rust/issues/111540)
+
+This adds a new profile setting to control how paths are sanitised in the resulting binary.
+This can be enabled like so:
+
+```toml
+cargo-features = ["trim-paths"]
+
+[package]
+# ...
+
+[profile.release]
+trim-paths = ["all"]
+```
+
+To set this in a profile in Cargo configuration, you need to use either
+`-Z trim-paths` or `[unstable]` table to enable it. For example,
+
+```toml
+# .cargo/config.toml
+[unstable]
+trim-paths = true
+
+[profile.release]
+rustflags = ["diagnostics"]
+```
+
+### Documentation updates
+
+#### trim-paths
+
+*as a new ["Profiles settings" entry](./profiles.html#profile-settings)*
+
+`trim-paths` is a profile setting which enables and controls the sanitisation of file paths in build outputs.
+It takes an array of the following values:
+
+- `"none"` and `false` --- disable path sanitisation
+- `"macro"` --- sanitise paths in the expansion of `std::file!()` macro.
+    This is where paths in embedded panic messages come from
+- `"diagnostics"` --- sanitise paths in printed compiler diagnostics
+- `"object"` --- sanitise paths in compiled executables or libraries
+- `"all"` and `true` --- sanitise paths in all possible locations
+
+It is defaulted to `none` for the `dev` profile, and `object` for the `release` profile.
```

Comment:
I don't think this was implemented, right?

Reply:
Thank you. It's a bit awkward to implement, as `dev` and `release profiles are kinda hard-coded. And we need to take care of both route of `Cargo.toml` and `.cargo/config.toml`. Anyway, this will be done before ready for review.